casper/src/main/scala/coop/rchain/casper/merging/DeployChainIndex.scala

```diff
@@ -83,4 +89,18 @@ object DeployChainIndex {
         Objects.hash(deployIds.map(id => DeployIdWithCost(id, 0)).map(_.id): _*)
       )
     }
+
+  def deployChainCost(r: DeployChainIndex) = r.deploysWithCost.map(_.cost).sum
+  def isDependency(a: DeployChainIndex, dependencyFor: DeployChainIndex) =
+    EventLogMergingLogic.depends(dependencyFor.eventLogIndex, a.eventLogIndex)
```

Comment:
Dependency direction is now computed in reverse - whether the first argument is a dependency for the second argument.